dataeng/resources/prefect-flows-deployment.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+set -ex
+
+# Creating python3.8 virtual env
+PLATFORM_VENV="platform_venv"
+virtualenv --python=python3.8 --clear "${PLATFORM_VENV}"
+source "${PLATFORM_VENV}/bin/activate"
+
+# Removing prefix 'prefect-flows-deployment-'
+# $FLOW will contain the name of flow
+FLOW=$(echo $JOB_NAME | cut -c 26-)
+
+cd $WORKSPACE/edx-prefectutils
+pip install -r requirements/base.txt
+pip install -r requirements/base.in
+
+# Install prefect-flow requirements
+cd $WORKSPACE/prefect-flows
+pip install -r requirements.txt
+
+
+
+# Get ECR authetication
+aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 216367352155.dkr.ecr.us-east-1.amazonaws.com
+
+# Check if this is first ever run of the job. If yes, then create ECR repository otherwise do not
+####### uncomment the following code after deploying all existing flows first time #######
+# if [ "$BUILD_NUMBER" -eq 1 ]
+# then
+#     echo "Creating repository as its a first build"
+#     aws ecr create-repository --repository-name $FLOW --region us-east-1
+# else
+#     echo "Not creating repository as its not a first build"
+# fi
+####### uncomment the above code after deploying all existing flows first time #######
+
```

Comment:
As per the comments above these lines are commented out for now but will be commented after all flows are run for the first time

Reply:
So I modified the command like 
```
# The following statement will help us to determine if repository already exists otherwise it will create a new repository with the name of flow
aws ecr describe-repositories --repository-names $FLOW --region us-east-1 || aws ecr create-repository --repository-name $FLOW --region us-east-1

```
where $FLOW is the name used with .py (e.g. `load-paypal-cases-to-snowflake-flow`) file and used in make command. The above command works fine but I realized our existing ecr repository name are not same of .py file rather they have - hyphens (e.g. `load-paypal-cases-to-snowflake-flow`) not sure why use this naming convention so this mismatch in the names is causing issue. I have also seen some repository name breaking this naming convention and have some different words in their name (e.g. `load-course-structure-to-snowflake-prefect-flow`)

Reply:
So when I am automating this this job script detects any toml or .py changes to either re-deploy (if its a change in existing flow) or create new repository and then deployment (if new flow added). My suggestion is to change the names of repository and be consistent with ecr repository names same as of .py file name, but I am looking for suggestions on this point

Reply:
I agree that standardization would be a good idea, but would renaming the repository be a problem in Prefect? Can it be done as a simple swap-in-place? Also, does repository name always equal flow name? I know that it does not have to be the same as the Python filename.

Reply:
Yes that is correct it doesn't have to be same as python filename, let me try to explain why I am suggesting it should be same as python file name.
Case 1: PR merged is a  New Flow added
- Suppose a person writing prefect-flow and name Docker storage image as different from its python file name, as I can see such naming in many of the flows currently [flow1](https://github.com/edx/prefect-flows/blob/master/flows/load_affiliate_window_transactions_to_snowflake/load_affiliate_window_transactions_to_snowflake.py#L103), [flow2](https://github.com/edx/prefect-flows/blob/master/flows/load_course_structure_to_snowflake/load_course_structure_to_snowflake.py#L612) and either we use decide to hyphens or underscore in the names if flow writer has changed the name we won't be able to check if the repository exits or not so that is why I feel standardization is important here.
Case 2: PR merged is a Existing Flow added
In this case will check the existence of repository on the basis of some naming convention hence some are already failing the naming convention so we will end up creating a new repository whereas their code will be pointing to old out updated ecr

Reply:
And if we have to standardize it think using the `_ underscore` to make is exactly like python file will simplify it otherwise it will be overhead to swap underscores with hyphens
@estute

Reply:
Here I am renaming the the flow names:
[https://github.com/edx/prefect-flows/pull/76](https://github.com/edx/prefect-flows/pull/76)